packages/components-js/tests/e2e/puppeteer/specs/flyout.e2e.ts

```diff
@@ -0,0 +1,639 @@
+import {
+  addEventListener,
+  createHTMLAttributes,
+  expectA11yToMatchSnapshot,
+  getActiveElementClassNameInShadowRoot,
+  getActiveElementId,
+  getActiveElementTagName,
+  getActiveElementTagNameInShadowRoot,
+  getAttribute,
+  getElementStyle,
+  getEventSummary,
+  getLifecycleStatus,
+  getProperty,
+  selectNode,
+  setContentWithDesignSystem,
+  setProperty,
+  waitForStencilLifecycle,
+} from '../helpers';
+import type { ElementHandle, Page } from 'puppeteer';
+import { Components } from '@porsche-design-system/components';
+import PFlyout = Components.PFlyout;
+
+let page: Page;
+const CSS_TRANSITION_DURATION = 600;
+const flyoutMinWidth = 320;
+
+beforeEach(async () => (page = await browser.newPage()));
+afterEach(async () => await page.close());
+
+const getHost = () => selectNode(page, 'p-flyout');
+const getFlyout = () => selectNode(page, 'p-flyout >>> .root');
+const getHeader = () => selectNode(page, 'p-flyout >>> .header');
+const getHeaderContent = () => selectNode(page, 'p-flyout >>> .header-content');
+const getHeaderSlottedContent = () => selectNode(page, '[slot="header"]');
+const getFooter = () => selectNode(page, 'p-flyout >>> .footer');
+const getFooterSlottedContent = () => selectNode(page, '[slot="footer"]');
+const getSecondaryContent = () => selectNode(page, 'p-flyout >>> .secondary-content');
+const getSecondaryContentSlottedContent = () => selectNode(page, '[slot="secondary-content"]');
+const getFlyoutDismissButton = () => selectNode(page, 'p-flyout >>> p-button-pure.dismiss');
+const getFlyoutDismissButtonReal = () => selectNode(page, 'p-flyout >>> p-button-pure.dismiss >>> button');
+const getBodyOverflow = async () => getElementStyle(await selectNode(page, 'body'), 'overflow');
+const getFlyoutVisibility = async () => await getElementStyle(await getFlyout(), 'visibility');
+const waitForFlyoutTransition = async () => {
+  await new Promise((resolve) => setTimeout(resolve, CSS_TRANSITION_DURATION));
+};
+const waitForSlotChange = () => new Promise((resolve) => setTimeout(resolve));
+
+const initBasicFlyout = (
+  flyoutProps: PFlyout = {
+    open: true,
+  },
+  flyoutSlots?: {
+    content?: string;
+    header?: string;
+    footer?: string;
+    secondaryContent?: string;
+  }
+): Promise<void> => {
+  const { header, content = '<p>Some Content</p>', footer, secondaryContent } = flyoutSlots || {};
+
+  const flyoutMarkup = `
+    <p-flyout ${createHTMLAttributes(flyoutProps)}>
+      ${header ? header : ''}
```

Comment:
Could initialize those variables with empty string and get rid of ternary expressions.